tests/api.c

```diff
@@ -16189,10 +16190,241 @@ static int test_wc_curve448_make_key (void)
 #endif
     return ret;    
 } /*END test_wc_curve448_make_key*/
+/*
+ * Testing test_wc_curve448_shared_secret_ex
+ */
+static int test_wc_curve448_shared_secret_ex (void) //ethan-3
+{
+    int ret = 0;
+#if defined(HAVE_CURVE448) 
+    WC_RNG        rng;
+    curve448_key  private_key, public_key;
+    byte          out[CURVE448_KEY_SIZE];
+    word32        outLen = sizeof(out);
+    int           endian = EC448_BIG_ENDIAN;
+    
+    printf(testingFmt, "wc_curve448_shared_secret_ex()");
+    
+    ret = wc_curve448_init(&private_key);
+    if (ret == 0) {
+        ret = wc_InitRng(&rng); 
+        if (ret == 0){
+            ret = wc_curve448_make_key(&rng, CURVE448_KEY_SIZE, &private_key);
+        }
+        if (wc_FreeRng(&rng) && ret == 0) {
+           ret = WOLFSSL_FATAL_ERROR;
+        }   
+    }
+    if (ret == 0){
+        ret = wc_curve448_init(&public_key);
+    }
+    if (ret == 0) {
+        ret = wc_InitRng(&rng); 
```

Comment:
I recommend initializing WC_RNG once at the beginning of the function, and freeing it once at the end instead of twice (one for each make_key call).